src/main.rs

```diff
@@ -142,6 +153,18 @@ fn main() -> Result<(), failure::Error> {
                 // stdout is buffered, flush it to see the effect immediately when hitting backspace
                 io::stdout().flush().ok();
 
+                if Instant::now() > token_expiry {
+                    // refresh token
+                    if let Some(new_token_info) = get_token(&mut oauth) {
+                        let (new_spotify, new_token_expiry) = get_spotify(new_token_info);
+                        spotify = new_spotify;
+                        token_expiry = new_token_expiry;
+                        app.spotify = Some(spotify);
+                    } else {
+                        // TODO: Handle error
```

Comment:
For now, on error let's just exit like this

```rust
println!("\nFailed to refresh authentication");
break;
```

The new line in the print is so the log will show in the console.